### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Yash-scaletech**
Add WFH Section

### CODE REVIEW

File: src/features/wfh/component/wfh.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+			const params = {
+				currentPage,
+				recordPerPage,
+				status: status ?? selectedStatus
+			};
+			setLoading(true);
+			httpService
+				.get(API_CONFIG.path.wfhList, params)
+				.then((res) => {
+					console.log('🚀 ~ file: wfh.tsx:52 ~ .then ~ res:', res.data.items);
```
Comment (daxesh-parmar-scaletech) [inline src/features/wfh/component/wfh.tsx]: remove logs

File: src/features/wfh/component/wfh.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+interface IWfhProps {
+	uId: string;
+}
+
+const Wfh: FC<IWfhProps> = ({ uId }) => {
+	const navigate = useNavigate();
+	const [isLoading, setLoading] = useState(false);
+	const [selectedStatus, setSelectedStatus] = useState<string>('');
+
+	const [wfhList, setWfhList] = useState<[]>([]);
```
Comment (daxesh-parmar-scaletech) [inline src/features/wfh/component/wfh.tsx]: give proper type

File: src/features/wfh/component/wfh.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+				</Flex>
+				<TableSelection
+					isLoading={isLoading}
+					userList={wfhList}
+					pagination={pagination}
+					handlePagination={handlePagination}
+					onRecordsPerPageChange={onRecordsPerPageChange}
+					columns={getLeaveColumns()}
+					leave={true}
+					isUpcomingWfh={true}
```
Comment (daxesh-parmar-scaletech) [inline src/features/wfh/component/wfh.tsx]: no need to write true

File: src/shared/components/table/container/table.tsx [hunk 4]
```diff
@@ -73,12 +74,13 @@ const useStyles = createStyles((theme) => {
...
-	const { isLoading, userList, columns, leave, className } = props;
+	const { isLoading, userList, columns, leave, isUpcomingWfh, className } = props;
 
 	const { classes } = useStyles();
 
-	const getRowClass = (isUpcomingLeave) => {
-		return leave && !isUpcomingLeave.isUpcomingLeave ? 'gray-background' : '';
+	const getRowClass = (item) => {
+		const upcomingProperty = isUpcomingWfh ? 'isUpcomingWfh' : 'isUpcomingLeave';
+		return leave && !item[upcomingProperty] ? 'gray-background' : '';
```
Comment (daxesh-parmar-scaletech) [inline src/shared/components/table/container/table.tsx]: use useMemo

**Review by daxesh-parmar-scaletech — CHANGES_REQUESTED** [review]

File: src/features/wfh/component/wfh.tsx (hunk 1 above)
Comment (Yash-scaletech) [reply src/features/wfh/component/wfh.tsx]: Done

File: src/features/wfh/component/wfh.tsx (hunk 3 above)
Comment (Yash-scaletech) [reply src/features/wfh/component/wfh.tsx]: Done

File: src/features/wfh/component/wfh.tsx (hunk 2 above)
Comment (Yash-scaletech) [reply src/features/wfh/component/wfh.tsx]: Done

File: src/shared/components/table/container/table.tsx (hunk 4 above)
Comment (Yash-scaletech) [reply src/shared/components/table/container/table.tsx]: Done

**Review by daxesh-parmar-scaletech — APPROVED** [review]

**Pull request merged**